telemetry_core/src/main/java/com/newrelic/telemetry/MetricBatchSenderFactory.java

```diff
@@ -4,18 +4,17 @@
  */
 package com.newrelic.telemetry;
 
+import com.newrelic.telemetry.http.HttpPoster;
 import com.newrelic.telemetry.metrics.MetricBatchSender;
 import com.newrelic.telemetry.metrics.MetricBatchSenderBuilder;
 import java.time.Duration;
 
 /**
- * A builder class for creating a MetricBatchSender that uses okhttp as the underlying http client
- * implementation.
+ * A factory interface for creating a MetricBatchSender.
  *
- * <p>Note: This class will be deprecated in the near future and be replaced with something named
- * more succinctly.
+ * <p>Concrete implementations use different HTTP providers.
  */
-public class SimpleMetricBatchSender {
```

Comment:
If we want to replace this (and I think we do!), we should definitely start by marking this class `@Deprecated` for at least some amount of time before we delete it altogether. There are users in the wild for whom this will be a breaking change.

Reply:
They're not deleted - I've converted them into interfaces, so I can sidestep the static state problem.